pose-detection/src/calculators/relative_velocity_filter.ts

```diff
@@ -0,0 +1,113 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+import {MACRO_SECONDS_TO_SECOND, SECOND_TO_MACRO_SECONDS} from './constants';
+import {WindowElement} from './interfaces/common_interfaces';
+import {VelocityFilterConfig} from './interfaces/config_interfaces';
+import {LowPassFilter} from './low_pass_filter';
+
+/**
+ * This filter keeps track (on a window of specified size) of value changes
+ * over time, which as result gives us velocity of how value changes over time.
+ * With higher velocity it weights new values higher.
+ *
+ * Use `windowSize` and `velocityScale` to tweak this filter for your use case.
+ */
+// ref:
+// https://github.com/google/mediapipe/blob/master/mediapipe/util/filtering/relative_velocity_filter.cc
+export class RelativeVelocityFilter {
+  private lastValue: number = 0;
+  private lastValueScale: number = 1;
+  private lastTimestamp: number = -1;
+
+  private window: WindowElement[] = [];
+  private lowPassFilter: LowPassFilter = new LowPassFilter(1.0);
+
+  /**
+   * Constructor of `RelativeVelocityFilter` class.
+   * @param config
+   *        `windowSize`:  Higher windowSize adds to lag and to stability.
+   *        `velocityScale`: Lower velocityScale adds to lag and to stability.
+   */
+  constructor(private readonly config: VelocityFilterConfig) {}
+
+  /**
+   * Applies filter to the value.
+   * @param value valueToFilter.
+   * @param macroSeconds timestamp associated with the value (for instance,
+   *     timestamp of the frame where you got value from).
+   * @param valueScale value scale (for instance, if your value is a distance
+   *     detected on a frame, it can look same on different devices but have
+   *     quite different absolute values due to different resolution, you
+   *     should come up with an appropriate parameter for your particular use
+   *     case).
+   */
+  apply(value: number, macroSeconds: number, valueScale: number): number {
+    if (value == null) {
+      return value;
+    }
+
+    const $macroSeconds = Math.trunc(macroSeconds);
+    if (this.lastTimestamp >= $macroSeconds) {
+      // Results are unpreditable in this case, so nothing to do but return
+      // same value.
+      return value;
+    }
+
+    let alpha;
+    if (this.lastTimestamp === -1) {
+      alpha = 1;
+    } else {
+      // Implement the DistanceEstimationMode.kLegacyTransition.
+      const distance =
+          value * valueScale - this.lastValue * this.lastValueScale;
```

Comment:
FYI, I see you used the legacy distance estimation mode. And that's what we use in our code. But probably we'll update